collector-http/src/main/java/zipkin2/collector/otel/http/OpenTelemetryHttpCollector.java

```diff
@@ -52,38 +78,87 @@ public static final class Builder extends CollectorComponent.Builder {
   }
 
   final Collector collector;
+
   final CollectorMetrics metrics;
 
   OpenTelemetryHttpCollector(Builder builder) {
     collector = builder.delegate.build();
     metrics = builder.metrics;
   }
 
-  @Override public OpenTelemetryHttpCollector start() {
+  @Override
+  public OpenTelemetryHttpCollector start() {
     return this;
   }
 
-  @Override public String toString() {
+  @Override
+  public String toString() {
     return "OpenTelemetryHttpCollector{}";
   }
 
   /**
    * Reconfigures the service per https://opentelemetry.io/docs/specs/otlp/#otlphttp-request
    */
-  @Override public void reconfigure(ServerBuilder sb) {
+  @Override
+  public void reconfigure(ServerBuilder sb) {
+    sb.decorator(DecodingService.newDecorator(StreamDecoderFactory.gzip()));
     sb.service("/v1/traces", new HttpService(this));
   }
 
   static final class HttpService extends AbstractHttpService {
+    private static final Logger LOG = Logger.getLogger(HttpService.class.getName());
+
     final OpenTelemetryHttpCollector collector;
 
     HttpService(OpenTelemetryHttpCollector collector) {
       this.collector = collector;
     }
 
-    @Override protected HttpResponse doPost(ServiceRequestContext ctx, HttpRequest req)
-        throws Exception {
-      throw new RuntimeException("Implement me!");
+    @Override
+    protected HttpResponse doPost(ServiceRequestContext ctx, HttpRequest req) {
+      CompletableCallback result = new CompletableCallback();
+      req.aggregate(AggregationOptions.usePooledObjects(ByteBufAllocator.DEFAULT, ctx.eventLoop()
+      )).handle((msg, t) -> {
+        if (t != null) {
+          result.onError(t);
+          return null;
+        }
+        try (HttpData content = msg.content()) {
+          if (content.isEmpty()) {
+            result.onSuccess(null);
+            return null;
+          }
+
+          try {
+            ExportTraceServiceRequest request = ExportTraceServiceRequest.parseFrom(CodedInputStream.newInstance(content.byteBuf().nioBuffer()));
```

Comment:
We could probably do:
```suggestion
            ExportTraceServiceRequest request = ExportTraceServiceRequest.parseFrom(UnsafeByteOperations.unsafeWrap(content.byteBuf().nioBuffer()).newCodedInput());
```